crates/adapters/src/transport/http/input.rs

```diff
@@ -88,20 +89,83 @@ impl HttpInputEndpointInner {
             status_notifier: watch::channel(()).0,
             details: Mutex::new(None),
             force,
+        });
+        tokio::spawn(HttpInputEndpointInner::background_task(
+            inner.clone(),
+            receiver,
+        ));
+        inner
+    }
+
+    async fn background_task(self: Arc<Self>, mut receiver: UnboundedReceiver<InputReaderCommand>) {
+        while let Some(message) = receiver.recv().await {
+            let _guard = info_span!("http_input").entered();
```

Comment:
I think it would be better to use [in_scope](https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.in_scope) instead of `entered` here because of [asynchronous behavior of enter(ed)](https://docs.rs/tracing/latest/tracing/span/struct.Span.html#in-asynchronous-code).